### PR TITLE
tests: add report/ to tsconfig

### PR DESCRIPTION
**Pull request by connorjclark**
Forgot to add this folder to the tsconfig during #12690 

Tests remain un-(type-)checked.

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -1,6 +1,5 @@
 **/node_modules/**
 **/third_party/**
-**/generated/**
```
Comment (connorjclark) [inline .eslintignore:3]: drive-by removal of `generated` which was actually cut before going to master

File: report/clients/treemap.js [hunk 2]
```diff
@@ -13,6 +13,7 @@ import {I18n} from '../renderer/i18n.js';
 import {Logger} from '../renderer/logger.js';
 import {TextEncoding} from '../renderer/text-encoding.js';
 
+/** @typedef {import('../../lighthouse-treemap/types/treemap')} */
```
Comment (connorjclark) [inline report/clients/treemap.js:16]: was pleasantly surprised this worked

EDIT: except it works too well.... this pollutes the `window` _everywhere_ in lighthouse-core,cli. oops. @brendankenny any ideas, or just ts-expect-error? These files are rather temporary if the report refactor works out...

File: .eslintignore (hunk 1 above)
Comment (adamraine) [reply .eslintignore:3]: Looks like this was hiding eslint errors in `lighthouse-core/lib/cdt/generated/SourceMap.js`. Maybe just make this rule specific to that directory?

**Review by adamraine — APPROVED** [review]
